packages/mocktail/lib/src/_register_matcher.dart

```diff
@@ -6,6 +6,7 @@ Map<Type, Object?> _createInitialFallbackValues() {
   final result = <Type, Object?>{};
 
   void createValue<T>(T value) {
+    assert(!result.containsKey(T));
```

Comment:
Why is this assertion needed?

Reply:
This is just a local helper inside the function to create default values, and we don't want to create default values twice as that would be a bug with our default value generation.

Reply:
Ah I see 👍
Can we also add a test for this?

Reply:
I'm not sure how that should work, do you have a suggestion?